consensus/src/liveness/proposal_generator.rs

```diff
@@ -383,7 +383,7 @@ impl ProposalGenerator {
 
             if !payload.is_direct()
                 && max_txns_from_block_to_execute.is_some()
-                && payload.len() as u64 > max_txns_from_block_to_execute.unwrap()
+                && payload.len() as u64 > max_txns_from_block_to_execute.unwrap_or(0)
```

Comment:
this should be consolidated to `max_txns_from_block_to_execute.map_or(false, |v| payload.len() as u64 > v)`